src/java/picard/analysis/artifacts/Transition.java

```diff
@@ -51,4 +46,46 @@ public Transition matchingRef() {
     public String toString() {
         return this.ref + ">" + this.call;
     }
+
+    protected enum Base {
+        A ('A'),
+        C ('C'),
+        G ('G'),
+        T ('T');
+
+        public byte base;
+
+        private Base(final char base) {
+            this.base = (byte)base;
+        }
+    }
+
+    // only 4 of the values will be used but we want to optimize for speed by accessing via the int value of a char
+    static protected final int[] baseIndexMap = new int[256];
+    static {
+        Arrays.fill(baseIndexMap, -1);
+        for (final Base b : Base.values()) {
+            baseIndexMap[b.base] = b.ordinal();
+        }
+    }
+
+    static private final Transition[][] transitionIndexMap = new Transition[Base.values().length][Base.values().length];
+    static {
+        transitionIndexMap[Base.A.ordinal()][Base.A.ordinal()] = AtoA;
```

Comment:
sorry, that might seem more snarky than I intended....
transitionOf uses the array we are trying to build here...